### PR TITLE
Update vm mapping documentation

### PR DESCRIPTION
**Pull request by rmohr**
Update VM mapping decision principles on how to extend the API.

Signed-off-by: Roman Mohr <rmohr@redhat.com>

### CODE REVIEW

File: docs/vm-configuration.md [hunk 1]
```diff
@@ -281,59 +281,49 @@ be taken into account:
...
     storage), which need to be fit into the existing structure.
 
-**Note**: Point **4** is still under debate.
+## Devices
 
-## General mapping rules
+### All devices need to be ordered the same way libvirt internally orders them
 
-### All devices need to live directly under *devices*
+Libvirt sorts devices into buckets based on their types. This means that if not
```
Comment (stu-gott) [inline docs/vm-configuration.md]: typo: "if no" vs "if not"

File: docs/vm-configuration.md [hunk 2]
```diff
@@ -281,59 +281,49 @@ be taken into account:
...
+PCI addresses are specified by hand, libvirt will always assign lower PCI
+addresses to disks than to video devices. No matter in which order they appear
+in the XML. The order withing devices of the same type is kept.
```
Comment (stu-gott) [inline docs/vm-configuration.md]: typo: "within"

File: docs/vm-configuration.md (hunk 1 above)
Comment (rmohr) [reply docs/vm-configuration.md]: done

File: docs/vm-configuration.md (hunk 2 above)
Comment (rmohr) [reply docs/vm-configuration.md]: done

**Pull request merged**